### PR TITLE
Fixed bug in the computation of the Peaceman index

### PR DESCRIPTION
**Pull request by francoishamon**
This PR fixes a bug in the computation of the well Peaceman index that occurs when the perforation is placed exactly at the center of the well element (my bad). It also fixes some well function names that were not following the camelCase convention. 

This PR does not require a rebaseline (the bug does not appear in the integrated tests).

### CODE REVIEW

File: src/coreComponents/meshUtilities/PerforationData.cpp [hunk 1]
```diff
@@ -78,8 +76,8 @@ void DecideWellDirection( VEC_TYPE const & vecWellElemCenterToPerf,
                           real64 & k2 )
 {
   // vertical well (approximately along the z-direction)
-  if( fabs( vecWellElemCenterToPerf[2] ) > fabs( vecWellElemCenterToPerf[0] )
-      && fabs( vecWellElemCenterToPerf[2] ) > fabs( vecWellElemCenterToPerf[1] ) )
+  if( fabs( vecTopToBottom[2] ) > fabs( vecTopToBottom[0] )
```
Comment (TotoGaz) [inline src/coreComponents/meshUtilities/PerforationData.cpp:79]: I would rename `vecTopToBottom` to `topToBottom`. Or at least `topToBottomVec`?

File: src/coreComponents/meshUtilities/PerforationData.cpp [hunk 2]
```diff
@@ -151,15 +150,21 @@ void PerforationData::computeWellTransmissibility( MeshLevel const & mesh,
...
 
-    // compute the vector perforation - well elem center
-    // this vector will be used to decide whether this is a vectical well or not
-    localIndex const wellElemIndex   = m_wellElementIndex[iperf];
-    real64 vecWellElemCenterToPerf[3] = LVARRAY_TENSOROPS_INIT_LOCAL_3( wellElemCenter[wellElemIndex] );
-    LvArray::tensorOps::subtract< 3 >( vecWellElemCenterToPerf, m_location[iperf] );
+    // get the index of the well element, and of the neighboring well nodes
+    localIndex const wellElemIndex = m_wellElementIndex[iperf];
+    localIndex const topNode = elemToNodeMap[wellElemIndex][InternalWellGenerator::NodeLocation::TOP];
+    localIndex const bottomNode = elemToNodeMap[wellElemIndex][InternalWellGenerator::NodeLocation::BOTTOM];
```
Comment (TotoGaz) [inline src/coreComponents/meshUtilities/PerforationData.cpp:156]: One funny test could be to swap the `0/1` values of `TOP` and `BOTTOM` to see if it still works. 😉

**Review by TotoGaz — APPROVED** [review]
LGTM

File: src/coreComponents/meshUtilities/PerforationData.cpp (hunk 1 above)
Comment (francoishamon) [reply src/coreComponents/meshUtilities/PerforationData.cpp]: Done, I renamed it to `topToBottomVec`

File: src/coreComponents/meshUtilities/PerforationData.cpp (hunk 2 above)
Comment (francoishamon) [reply src/coreComponents/meshUtilities/PerforationData.cpp:156]: I swapped the 0 and 1 (locally, not in the version I committed) and it worked. The integrated tests pass, except for the node list of the `WellElementSubRegion` that is obsviously different if 0 and 1 are swapped.

As a side note, I noticed that I broke an integrated test `deadoil_3ph_staircase_3d` with #901, I will fix it just after this PR using the Dead-Oil branch.

**Pull request merged**